app/controllers/peer_reviews_controller.rb

```diff
@@ -107,18 +115,21 @@ def download_reviewer_reviewee_mapping
 
   def csv_upload_handler
     assignment_id = params[:assignment_id]
+    pr_mapping = params[:peer_review_mapping]
+    encoding = params[:encoding]
 
     if params[:peer_review_mapping].nil?
       flash_message(flash[:error], I18n.t('csv.group_to_grader'))
     else
-      result = MarkusCSV.parse(params[:peer_review_mapping].read,
-                               encoding: params[:encoding]) do |row|
-        raise CSVInvalidLineError if row.empty?
-        result = Result.find(row.first)
-        reviewer = Grouping.joins(:group).find_by(
-                                groups: { group_name: row.second },
-                                assignment_id: assignment_id)
-        PeerReview.create!(result: result, reviewer: reviewer)
+      result = MarkusCSV.parse(pr_mapping.read, encoding: encoding) do |row|
+        raise CSVInvalidLineError if row.size < 2
+        reviewer = Grouping.joins(:group).find_by(groups: { group_name: row.first })
```

Comment:
This logic isn't quite right if the same group has groupings for multiple assignments.

First get the `Group`, then use the `grouping_for_assignment` method to get the right grouping. Applies below as well.